db/migrate/20150716133712_add_unique_index_on_journals.rb

```diff
@@ -0,0 +1,171 @@
+    end
+
+    unless records_equivalent?(a.data, b.data)
+      say 'Difference found in related data table (e.g. work_package_journals)', subitem: true
+      return false
+    end
+
+    a_attachments = a.attachable_journals.pluck(:attachment_id).sort
+    b_attachments = b.attachable_journals.pluck(:attachment_id).sort
+    unless a_attachments == b_attachments
+      say 'Difference found in attachable_journals', subitem: true
+      return false
+    end
+
+    a_custom_fields = customizable_journals_to_hash a.customizable_journals
+    b_custom_fields = customizable_journals_to_hash b.customizable_journals
+    unless a_custom_fields == b_custom_fields
+      say 'Difference found in customizable_journals', subitem: true
+      return false
+    end
+
+    true
+  end
+
+  def records_equivalent?(a, b)
+    if a.nil? || b.nil?
+      return a == b
+    end
+
+    ignored = [:id, :journal_id, :created_at, :updated_at]
+    a.attributes.symbolize_keys.except(*ignored) == b.attributes.symbolize_keys.except(*ignored)
+  end
+
+  def customizable_journals_to_hash(customizable_journals)
+    customizable_journals.inject({}) { |hash, custom_journal|
+      hash[custom_journal.custom_field_id] = custom_journal.value
+      hash
+    }
+  end
+
+  def abort_on_undeleted_pairs(undeleted_pairs)
+    return unless undeleted_pairs.any?
+
+    say '', subitem: true
+    say 'There were journals that had a duplicate, but were not deleted.', subitem: true
+    say 'You have to manually decide how to proceed with these journals.', subitem: true
+    say 'Please compare the corresponding entries in the following tables:', subitem: true
+    say ' * journals', subitem: true
+    say ' * attachable_journals', subitem: true
+    say ' * customizable_journals', subitem: true
+    say ' * {type}_journals, with {type} being indicated by the journable_type', subitem: true
+    say '', subitem: true
+    say 'The following table lists the remaining duplicate pairs,', subitem: true
+    say 'note that only one entry per pair is supposed to be deleted:', subitem: true
+
+    column_width = 20
+    say '-' * (column_width *  2 + 7), subitem: true
+    say "| #{'journal 1'.rjust(column_width)} | #{'journal 2'.rjust(column_width)} |", subitem: true
+    say '-' * (column_width *  2 + 7), subitem: true
+    undeleted_pairs.each do |undeleted_id, duplicate_id|
+      say "| #{undeleted_id.to_s.rjust(column_width)} | #{duplicate_id.to_s.rjust(column_width)} |",
+          subitem: true
+    end
+    say '-' * (column_width *  2 + 7), subitem: true
+
+    raise "Can't continue migration safely because of duplicate journals!"
+  end
+
+  # Using a custom (light weight) implementation of Journal here, because we don't know
+  # how the original might change in the future. Changes could potentially break our untested
+  # migrations. By providing a minimal custom implementation, I hope to reduce that risk.
+  class MigrationHelperJournal < ActiveRecord::Base
```

Comment:
If you are doing this for the Journals themselves, what is the reason for not doing it with Attachable/Customizable/[Journable a to z]Journal?

Seems inconsistent to me.